common/utils/src/main/java/org/graalvm/buildtools/agent/AgentConfiguration.java

```diff
@@ -144,23 +146,43 @@ private void addDefaultAccessFilter() {
 
         String tempDir = System.getProperty("java.io.tmpdir");
         Path agentDir = Path.of(tempDir).resolve("agent-config");
-        Path accessFilterFile = agentDir.resolve("access-filter.json");
+        Path accessFilterFile = agentDir.resolve(ACCESS_FILTER_PREFIX + ACCESS_FILTER_SUFFIX);
         if (Files.exists(accessFilterFile)) {
             accessFilterFiles.add(accessFilterFile.toString());
             return;
         }
 
-        try(InputStream accessFilter = AgentConfiguration.class.getResourceAsStream(DEFAULT_ACCESS_FILTER_FILE)) {
-            if (accessFilter != null) {
-                if (!Files.exists(agentDir)) {
-                    Files.createDirectory(agentDir);
-                }
+        try(InputStream accessFilterData = AgentConfiguration.class.getResourceAsStream(DEFAULT_ACCESS_FILTER_FILE_LOCATION)) {
+            if (accessFilterData == null) {
+                throw new IOException("Cannot find access-filter.json on default location: " + DEFAULT_ACCESS_FILTER_FILE_LOCATION);
+            }
+
+            try {
+                Files.createDirectory(agentDir);
+            } catch (FileAlreadyExistsException e) {
+                logger.info("Ignore directory creation because " + agentDir + " directory is already created.");
+            }
 
-                Files.copy(accessFilter, accessFilterFile, StandardCopyOption.REPLACE_EXISTING);
-                accessFilterFiles.add(accessFilterFile.toString());
-            } else {
-                throw new IOException("Cannot find access-filter.json on default location: " + DEFAULT_ACCESS_FILTER_FILE);
+            Path tmpAccessFilter = null;
+            try {
+                 tmpAccessFilter = Files.createTempFile(agentDir, ACCESS_FILTER_PREFIX, ACCESS_FILTER_SUFFIX);
```

Comment:
You mentioned that his gave you:
```
Caused by: java.nio.file.FileAlreadyExistsException: /tmp/agent-config/access-filter21830177911714255.json
```
Why is this not an issue anymore?

Reply:
Maybe it's better to create the temp name differently. A combination of PID + UTC timestamp would be good.